File: src/Riada.Domain/Entities/Membership/Member.cs
using Riada.Domain.Enums;

namespace Riada.Domain.Entities.Membership;

public class Member
{
    public uint Id { get; set; }
    public string LastName { get; set; } = null!;
    public string FirstName { get; set; } = null!;
    public string Email { get; set; } = null!;
    public Gender Gender { get; set; } = Gender.Unspecified;
    public DateOnly DateOfBirth { get; set; }
    public string Nationality { get; set; } = "Belgian";
    public string? MobilePhone { get; set; }
    public string? AddressStreet { get; set; }
    public string? AddressCity { get; set; }
    public string? AddressPostalCode { get; set; }
    public MemberStatus Status { get; set; } = MemberStatus.Active;
    public uint? ReferralMemberId { get; set; }
    public PrimaryGoal? PrimaryGoal { get; set; }
    public AcquisitionSource? AcquisitionSource { get; set; }
    public bool MedicalCertificateProvided { get; set; }
    public DateTime GdprConsentAt { get; set; }
    public bool MarketingConsent { get; set; }
    public DateOnly? LastVisitDate { get; set; }
    public uint TotalVisits { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation
    public Member? ReferralMember { get; set; }
    public ICollection<Member> ReferredMembers { get; set; } = [];
    public ICollection<Contract> Contracts { get; set; } = [];
}

File: dashstack(1)/src/app/types/member.ts
export interface MemberSummary {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  status: "Active" | "Suspended" | "Anonymized";
  currentPlan: string | null;
  homeClub: string | null;
  lastVisitDate: string | null;
  totalVisits: number;
  riskScore: number;
  mobilePhone?: string;
}

export interface ContractDetail {
  id: number;
  planName: string;
  homeClub: string;
  startDate: string;
  endDate: string | null;
  contractType: "FixedTerm" | "OpenEnded";
  status: "Active" | "Suspended" | "Expired" | "Cancelled";
  freezeStartDate: string | null;
  freezeEndDate: string | null;
  activeOptions: string[];
}

export interface MemberNote {
  author: string;
  date: string;
  text: string;
}

export interface TimelineEvent {
  id: number;
  date: string;
  time: string;
  type: string;
  title: string;
  desc?: string;
  icon: any;
  color: string;
  bg: string;
}

export interface MemberDetailType {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  mobilePhone: string | null;
  status: "Active" | "Suspended" | "Anonymized";
  gender: "Male" | "Female" | "Unspecified";
  dateOfBirth: string;
  nationality: string;
  addressStreet: string | null;
  addressCity: string | null;
  addressPostalCode: string | null;
  primaryGoal: "WeightLoss" | "MuscleGain" | "Fitness" | "Maintenance" | "Other" | null;
  acquisitionSource: "WebAdvertising" | "SocialMedia" | "WordOfMouth" | "Other" | null;
  medicalCertificateProvided: boolean;
  gdprConsentAt: string;
  marketingConsent: boolean;
  referralMemberName: string | null;
  totalVisits: number;
  lastVisitDate: string | null;
  riskScore: number;
  weeklyFrequency: string;
  engagementLevel: string;
  createdAt: string;
  contracts: ContractDetail[];
  alerts: string[];
  notes: MemberNote[];
  timeline: TimelineEvent[];
  recentVisits: { day: string; visits: number }[];
  pendingInvoicesCount: number;
  totalPaid: number;
  sponsoredMembersCount: number;
}
